consensus/istanbul/backend/engine_test.go

```diff
@@ -73,7 +75,46 @@ func newBlockChain(n int) (*core.BlockChain, *backend) {
 	return blockchain, b
 }
 
-func getGenesisAndKeys(n int) (*core.Genesis, []*ecdsa.PrivateKey) {
+// in this test, we can set n to 1, and it means we can process Istanbul and commit a
+// block by one node. Otherwise, if n is larger than 1, we have to generate
+// other fake events to process Istanbul.
+func newLegacyBlockChain(n int) (*core.BlockChain, *backend) {
+	genesis, nodeKeys := getGenesisAndKeys(n, false)
+	memDB := rawdb.NewMemoryDatabase()
+	config := istanbul.DefaultConfig
+	// Use the first key as private key
+	b, _ := New(config, nodeKeys[0], memDB).(*backend)
+
+	config.QbftBlock = nil
```

Comment:
does this end up changing the `defaultConfig` for subsequent tests (we are changing param of package-level struct pointer)?

Reply:
Creating a copy of the config and acting on it instead of DefaultConfig